packages/@aws-cdk/aws-apprunner/lib/service.ts

```diff
@@ -0,0 +1,722 @@
+import * as ecr from '@aws-cdk/aws-ecr';
+import { DockerImageAsset } from '@aws-cdk/aws-ecr-assets';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnService } from './';
+
+/**
+ * The image repository types
+ */
+export enum ImageRepositoryType {
+  /**
+   * Amazon ECR Public
+   */
+  ECR_PUBLIC = 'ECR_PUBLIC',
+
+  /**
+   * Amazon ECR
+   */
+  ECR = 'ECR',
+}
+
+/**
+ * The number of CPU units reserved for each instance of your App Runner service.
+ */
+export class Cpu {
+  /**
+   * 1 vCPU
+   */
+  public static readonly ONE_VCPU = Cpu.of('1 vCPU')
+
+  /**
+   * 2 vCPU
+   */
+  public static readonly TWO_VCPU = Cpu.of('2 vCPU')
+
+  /**
+   * Custom CPU unit
+   *
+   * @param unit custom CPU unit
+   */
+  public static of(unit: string) { return new Cpu(unit); }
+
+  /**
+   *
+   * @param unit The unit of CPU.
+   */
+  private constructor(public readonly unit: string) {}
+}
+
+
+/**
+ * The amount of memory reserved for each instance of your App Runner service.
+ */
+export class Memory {
+  /**
+   * 2 GB
+   */
+  public static readonly TWO_GB = Memory.of('2 GB')
+
+  /**
+   * 3 GB
+   */
+  public static readonly THREE_GB = Memory.of('3 GB')
+
+  /**
+   * 4 GB
+   */
+  public static readonly FOUR_GB = Memory.of('4 GB')
+
+  /**
+   * Custom Memory unit
+   *
+   * @param unit custom Memory unit
+   */
+  public static of(unit: string) { return new Memory(unit); }
+
+  /**
+   *
+   * @param unit The unit of memory.
+   */
+  private constructor(public readonly unit: string) { }
+}
+
+/**
+ * The code runtimes
+ */
+export enum CodeRuntime {
+  /**
+   * NodeJS 12
+   */
+  NODEJS_12 = 'NODEJS_12',
+
+  /**
+   * Python 3
+   */
+  PYTHON_3 = 'PYTHON_3',
+}
```

Comment:
If/when new runtimes get added, every user will have to upgrade to get them (or use escape hatches). Might be best to make an enum-like class with `CodeRuntime.fromName`.

Also.... Maybe go Lambda-like and simply call this `Runtime`?

Reply:
Makes sense. Let's call it `Runtime` with enum-like class.